### PR TITLE
View transitions docs

### PR DESCRIPTION
**Pull request by matthewp**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes https://github.com/withastro/docs/issues/3625
- New guide for experiment client-side routing. The approach is:
  - Describe how to enable
  - Describe how to use
  - Describe how to customize
  - Describe fallback choices

### CODE REVIEW

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 1]
```diff
@@ -0,0 +1,192 @@
+---
+title: Client-side routing (experimental)
+description: >-
+  How to enable experimental support for client-side routing in your Astro site.
+i18nReady: true
+---
+
+Support for client-side routing in Astro projects can be enabled behind an experimental flag. Client-side routing in Astro is powered by a new browser API, [view transitions](https://developer.chrome.com/docs/web-platform/view-transitions/) with fallback options for non-supporting browsers.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:8]: I feel we're being a bit redundant here, maybe reword it like this?

Also, as we're naming an API, I suppose it should be treated as a proper noun?

```suggestion
Support for client-side routing in Astro projects can be enabled behind an experimental flag. It is powered by the new [View Transitions](https://developer.chrome.com/docs/web-platform/view-transitions/) browser API, with fallback options for non-supporting browsers.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+Using our new view transitions support you can have seamless animations between pages. Astro comes with:
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:10]: We just said it's new above, so I would avoid saying it twice, plus the proper noun situation.
```suggestion
Using the View Transitions support you can have seamless animations between pages. Astro comes with:
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+- A component `<ViewTransitions />` that does the routing for you.
+- A few built-in animations, such as `slide` and `fade`.
+- Support for both forwards and backwards navigation animations.
+- The ability to customize animations any way you like.
+- Fallback support for browser that do not yet support view transition APIs.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:16]: I believe reading as "A ViewTransitions component" reads better than "A component ViewTransitions".

Typo `browser` -> `browsers`
```suggestion
- A `<ViewTransitions />` component that does the routing for you.
- A few built-in animations, such as `slide` and `fade`.
- Support for both forwards and backwards navigation animations.
- The ability to customize animations any way you like.
- Fallback support for browsers that do not yet support the View Transition APIs.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+:::caution
+Client-side routing is an experimental feature enabled in Astro 2.9. The API is subject to change before it is marked as stable.
+:::
+
+## Enabling Client-side Routing in your Project
+
+You can enable support for client-side routing through the `viewTransitions` experimental flag in your Astro config:
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:24]: Same thought about the reading order here.
```suggestion
You can enable support for client-side routing through the experimental `viewTransitions` flag in your Astro config:
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+Enabling this flag gives you access to the `ViewTransitions` component, as well as the ability to use `transition:animate` and other directives in your Astro components. Without this flag enabled, these features will be ignored.
+
+## Using Client-side Routing on a Page
+
+Astro does not automatically make your entire site into a SPA (Single-page App). Instead by default all pages on an Astro site get regular browser navigation. You can enable client-side routing on a per-page basis by importing the `ViewTransitions` component and placing it inside of your head.
+
+If you'd like to enable it on your entire site, you can do so by putting it in a reused component, such as a layout component or a commonly used head component, as are common conventions in Astro sites.
+
+For example, you might have a head component which you can update with:
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:44]:
```suggestion
For example, you might have a head component that you can update with:
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+<!-- Primary Meta Tags -->
+<title>{title}</title>
+<meta name="title" content={title} />
+<meta name="description" content={description} />
+
+<ViewTransitions />
+```
+
+Just by doing that, now the next page that you navigate to will happen without full-page navigation. Next you'll likely want to enable animations between common parts of your page.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:61]: comma nit
```suggestion
Just by doing that, now the next page that you navigate to will happen without full-page navigation. Next, you'll likely want to enable animations between common parts of your page.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    <main transition:animate="slide">
+      ...
+    </main>
+  </body>
+</html>
+```
+
+The `transition:animate` directive takes a string value, which is the name of the built-in animate, or an object for custom animations (described later). The built-in animations are:
+
+- `slide`: An animation where the old content slides out to the left and new content slides in from the right. On backwards navigation the animations are the opposite.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:88]: comma nit: strikes back
```suggestion
- `slide`: An animation where the old content slides out to the left and new content slides in from the right. On backwards navigation, the animations are the opposite.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+- `fade`: A cross-fade where the old content fades out and the new content fades in.
+- `morph`: This is the default where the browser determines the best way to animate the element depending on how similar the pages are. One use-case for this animation is if the element is positionally different between pages; it will appear to float to its new position. Another is if the element is in the exact same position; it will appear to not move at all.
+
+Additionally all of these animations can be customized by importing them through the `astro:transitions` namespace. See the __Customizing Animations__ section to learn how to use those.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:92]: return of the comma nit plus adding a reference link!
```suggestion
Additionally, all of these animations can be customized by importing them through the `astro:transitions` namespace. See the [Customizing Animations section](#customizing-animations) to learn how to use those.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+### Naming a transition
+
+Transitioning elements is built on the concept of a `view-transition-name`. An element that is in both the old and new page can be animated by giving them a common name.
+
+Astro will automatically give elements a `view-transition-name` that's based on the component it is in and where it is placed in the DOM. Most of the time this will be the same name between pages, which allows you not to worry about giving distinct names.
+
+In some cases an element by be positionally very different, in which case Astro's auto-naming will be wrong. You can specify a name for an element using `transition:name` directive, which will ensure it animates properly.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:100]: typo
```suggestion
In some cases an element by being positionally very different, in which case Astro's auto-naming will be wrong. You can specify a name for an element using `transition:name` directive, which will ensure it animates properly.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+	new: TransitionAnimation | TransitionAnimation[];
+}
+
+export interface TransitionDirectionalAnimations {
+	forwards: TransitionAnimationPair;
+	backwards: TransitionAnimationPair;
+}
+```
+
+A fully example of the fade animation is:
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:143]: typo
```suggestion
A full example of the fade animation is:
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+	backwards: anim,
+};
+---
+
+<header transition:animate={myFade}> ... </header>
+```
+
+## Fallback control
+
+The `<ViewTransitions />` router works best in browser that support view transitions (Chromium browsers a the moment), but also includes fallback support for other browsers. Even if the browser does not support view transitions you will still get in-browser navigation.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:173]: Tried to reduce the redundancy and changing browser to browsers
```suggestion
The `<ViewTransitions />` router works best in browsers that do support View Transitions (Chromium browsers a the moment), but also includes fallback support for other browsers. Even if the browser does not support it, you will still get in-browser navigation.
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+<title>My site</title>
+
+<ViewTransitions fallback="swap">
+```
+
+The possible values are:
+
+- `animate` - This is the default. Astro will simulate view transitions using custom attributes before swapping the page.
+- `swap` - Astro will not attempt to animate the page. Instead the old page will simply be swapped for the new one.
+- `none` - Astro will not do client-side routing at all. Instead you will get full page navigation in non-supporting browsers.
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:192]: comma nits
```suggestion
- `swap` - Astro will not attempt to animate the page. Instead, the old page will simply be swapped for the new one.
- `none` - Astro will not do client-side routing at all. Instead, you will get full page navigation in non-supporting browsers.
```

**Review by Yan-Thomas — COMMENTED** [review]
Yay, View Transitions support is 🔥, good job in the docs, I added a few suggestions!

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 13]
```diff
@@ -0,0 +1,192 @@
+---
+title: Client-side routing (experimental)
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:2]: nit: to match with the assets page `Assets (Experimental)`
```suggestion
title: Client-side routing (Experimental)
```

File: src/i18n/en/nav.ts [hunk 14]
```diff
@@ -83,6 +83,7 @@ export default [
 	{ text: 'Data Fetching', slug: 'guides/data-fetching', key: 'guides/data-fetching' },
 	{ text: 'Middleware', slug: 'guides/middleware', key: 'guides/middleware' },
 	{ text: 'Testing', slug: 'guides/testing', key: 'guides/testing' },
+	{ text: 'Client-side Routing (experimental)', slug: 'guides/client-side-routing', key: 'guides/client-side-routing'},
```
Comment (Yan-Thomas) [inline src/i18n/en/nav.ts:86]: same here
```suggestion
	{ text: 'Client-side Routing (Experimental)', slug: 'guides/client-side-routing', key: 'guides/client-side-routing'},
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 15]
```diff
@@ -58,11 +82,22 @@ import { ViewTransitions } from 'astro:transitions';
 <ViewTransitions />
 ```
 
-Just by doing that, now the next page that you navigate to will happen without full-page navigation. Next, you'll likely want to enable animations between common parts of your page.
+## Transition Directives
+
+Use `transition:*` directives on page elements in your `.astro` components to control the transition animation for individual elements on your page during navigation.
+
+- `transition:animate`: Defines the type of animation. Use Astro's [built-in animation directives](built-in-animation-directives) or [create custom transition animations](#customizing-animations).
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:89]: Fix link
```suggestion
- `transition:animate`: Defines the type of animation. Use Astro's [built-in animation directives](#built-in-animation-directives) or [create custom transition animations](#customizing-animations).
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 16]
```diff
@@ -0,0 +1,219 @@
...
+Use client-side routing in Astro with the `<ViewTransitions />` routing component and transition directives on a per-page basis, or site-wide.
+:::
+
+## Client-side Routing on a Page
+
+Import the `<ViewTransitions />` component and place it inside of a page's `<head>` to opt that page in to client-side routing on a per-page basis. You can then control the animation of page elements during navigation with [transition directives](#transition-directives):
+
+```astro title="src/pages/MyClientSideRoutingPage.astro"
+---
+import { ViewTransitions } from 'astro/components';
```
Comment (sarah11918) [inline src/content/docs/en/guides/client-side-routing.mdx:49]:
```suggestion
import { ViewTransitions } from 'astro:transitions';
```

File: src/content/docs/en/guides/client-side-routing.mdx [hunk 17]
```diff
@@ -44,7 +42,7 @@ Use client-side routing in Astro with the `<ViewTransitions />` routing componen
 
 ## Client-side Routing on a Page
 
-Import the `<ViewTransitions />` component and place it inside of a page's head to opt that page in to client-side routing on a per-page basis. You can additionally control the animation of page elements during navigation with [transition directives](#transition-directives):
+Import the `<ViewTransitions />` component and place it inside of a page's `<head>` to opt that page in to client-side routing on a per-page basis. You can then control the animation of page elements during navigation with [transition directives](#transition-directives):
```
Comment (Yan-Thomas) [inline src/content/docs/en/guides/client-side-routing.mdx:45]: Guess here it should be "into"?
```suggestion
Import the `<ViewTransitions />` component and place it inside of a page's `<head>` to opt that page into client-side routing on a per-page basis. You can then control the animation of page elements during navigation with [transition directives](#transition-directives):
```

Comment (sarah11918) [reply src/content/docs/en/guides/client-side-routing.mdx:45]: Interesting question! I was thinking of it from the "opt in" phrasing perspective, and while we probably could get away with either one, and language evolving yadda yadda yadda... I think *technically* and *pedantically* mine is considered traditionally correct? And maybe the jury is out on universally accepting "into"?

We're not moving "into" client-side routing. We're opting in... to what? To client-side routing? A quick check of a few sources isn't conclusive re: absolutely correct, but everyone's take seems to lean towards mine.  :smile:

Comment (Yan-Thomas) [reply src/content/docs/en/guides/client-side-routing.mdx:45]: Yeah, this one is confusing, I'm good with going with your version!

Comment (matthewp) [reply src/content/docs/en/guides/client-side-routing.mdx:45]: I don't think I understand this thread quite... happy with whatever you all decide here.
